### PR TITLE
Improve Bach docs

### PR DESCRIPTION
**Pull request by jansentom**

### CODE REVIEW

File: bach/bach/dataframe.py [hunk 1]
```diff
@@ -41,25 +42,74 @@ class DataFrame:
...
     A mutable DataFrame representing tabular data in a database and enabling operations on that data.
 
+    A Bach DataFrame object can be used to process large amounts of data on a database, while using an api
+    that is based on the pandas api. This allows the database to group and aggregate data, sample data and
+    do other operations that are not suitable for in memory processing. At any time it is possible to write
+    your Bach DataFrame to a pandas DataFrame.
+
+    Usage
+    ---------
+    It should generally not be required to construct Series instances manually. A DataFrame can be constructed
```
Comment (jansentom) [inline bach/bach/dataframe.py:52]: 'It should generally not be required to construct Series instances manually.' A bit strange to start the Usage of DataFrame with that you should not use Series?

File: bach/bach/dataframe.py [hunk 2]
```diff
@@ -41,25 +42,74 @@ class DataFrame:
...
+    using the any of the bach classmethods like from_table, from_model, or from_pandas. The returned DataFrame
+    can be thought of as a dict-like container for Bach Series objects.
+
+    Getting & Setting columns
+    ---------
```
Comment (jansentom) [inline bach/bach/dataframe.py:57]: The length of the dashes should match the length of the heading above

File: bach/bach/dataframe.py [hunk 3]
```diff
@@ -41,25 +42,74 @@ class DataFrame:
...
+    --------
+    df['a'] = df.column_name + 5
+    df['b'] = ''
+
+    If a Series of DataFrames do not share the same base node, it is possible to combine the data using
+    merge().
+
+    See also
+    ---------
+    Series.isin
```
Comment (jansentom) [inline bach/bach/dataframe.py:91]: Copy paste error?

File: bach/bach/dataframe.py [hunk 4]
```diff
@@ -41,25 +42,74 @@ class DataFrame:
...
-    on the data will result in SQL queries that build on top of the query of the base_node. The
-    index and series parameters contain meta information about the data in the base_node.
-
     The API of this DataFrame is partially compatible with Pandas DataFrames. For more on Pandas
     DataFrames see https://pandas.pydata.org/docs/reference/frame.html
     """
-
+    # todo note on ordering for slices?
+    # todo 'meaning that they originate from the same data source' ok, by approximation
+    # todo _get_dtypes also queries the database
```
Comment (jansentom) [inline bach/bach/dataframe.py:112]: but no volumes of data, just some metadata

File: bach/bach/dataframe.py [hunk 5]
```diff
@@ -41,25 +42,74 @@ class DataFrame:
...
-    contains. That can be a simple query on a table, but also a complicated query in itself. Operations
-    on the data will result in SQL queries that build on top of the query of the base_node. The
-    index and series parameters contain meta information about the data in the base_node.
-
     The API of this DataFrame is partially compatible with Pandas DataFrames. For more on Pandas
     DataFrames see https://pandas.pydata.org/docs/reference/frame.html
     """
-
+    # todo note on ordering for slices?
+    # todo 'meaning that they originate from the same data source' ok, by approximation
```
Comment (jansentom) [inline bach/bach/dataframe.py:111]: originate from the same node in the data graph? Thijs will probably explain in a bit more detail what is happening under the hood

File: bach/bach/dataframe.py [hunk 6]
```diff
@@ -41,25 +42,74 @@ class DataFrame:
...
-    The initial data of the DataFrame is the result of the SQL query that the `base_node` parameter
-    contains. That can be a simple query on a table, but also a complicated query in itself. Operations
-    on the data will result in SQL queries that build on top of the query of the base_node. The
-    index and series parameters contain meta information about the data in the base_node.
-
     The API of this DataFrame is partially compatible with Pandas DataFrames. For more on Pandas
     DataFrames see https://pandas.pydata.org/docs/reference/frame.html
     """
-
+    # todo note on ordering for slices?
```
Comment (jansentom) [inline bach/bach/dataframe.py:110]: Yeah, think that's a good idea. See Series for inspiration: https://github.com/objectiv/objectiv-analytics/pull/283/files#diff-bb3ff89f2adc9d945d0a5644475cb83e83f2c7c30123e818c1c9a058e9a8f05dR46

File: bach/docs/source/index.rst [hunk 7]
```diff
@@ -1,27 +1,80 @@
...
+a Series object:
+ * :py:meth:`~bach.dataframe.DataFrame.to_pandas()`
+ * :py:meth:`~bach.dataframe.DataFrame.head()`
+ * The property accessors :py:attr:`~bach.series.series.Series.value` (Series only)
+   , :py:attr:`~bach.dataframe.DataFrame.values`, and :py:attr:`~bach.dataframe.DataFrame.array`
+
+Typical usage would be to do all heavy lifting inside the database, and only query the aggregated/summarized
+output.
+
+Compatibility with pandas
```
Comment (thijs-obj) [inline bach/docs/source/index.rst:48]: I added this, but not sure how important it is

File: bach/docs/source/index.rst [hunk 8]
```diff
@@ -1,27 +1,80 @@
 .. _bach:
 
-Welcome to Bach's documentation!
-==================================
+Data Modeling
+=============
 
-Bach is Objectiv's data modeling library. With Bach, you can compose models with familiar Pandas-like dataframe operations in your notebook. It uses an SQL abstraction layer that enables models to run on the full dataset, and you can output models to SQL with a single command. It includes a set of operations that enable effective feature creation for datasets that embrace the open taxonomy of analytics.
```
Comment (thijs-obj) [inline bach/docs/source/index.rst:6]: I didn't change much here yet, only split it over multiple lines. We do need to talk about 'data modeling' here a bit first, as that's the heading on our website. So if we start directly about Bach that would be a bit odd.

File: bach/bach/dataframe.py (hunk 4 above)
Comment (hendrik-obj) [reply bach/bach/dataframe.py:111]: for bigquery it means the entire table gets queried. is that different for postgres?

**Review by jansentom — COMMENTED** [review]
Everything but generate.py is ready to go IMHO. Needs to be merged just before https://github.com/objectiv/objectiv-analytics/pull/287/files to not break main @borft

**Pull request merged**